ark/analysis/dimensionality_reduction.py

```diff
@@ -31,52 +88,31 @@ def visualize_dimensionality_reduction(cell_data, columns, category, color_map="
         raise ValueError(f"The algorithm specified must be one of the following: "
                          f"{['UMAP', 'PCA', 'tSNE']}")
 
+    graph_title = "%s projection of data" % algorithm
+
     if algorithm == "UMAP":
         reducer = umap.UMAP()
 
         column_data = cell_data[columns].values
         scaled_column_data = StandardScaler().fit_transform(column_data)
         embedding = reducer.fit_transform(scaled_column_data)
 
-        fig1 = plt.figure(1)
-        sns.scatterplot(x=embedding[:, 0], y=embedding[:, 1], hue=cell_data[category],
-                        palette=color_map, data=cell_data, legend="full", alpha=0.3)
-        plt.legend(bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.)
-        plt.title('UMAP projection of data', fontsize=24)
-        fig1.show()
-        if save_dir is not None:
-            plt.savefig(os.path.join(save_dir, "UMAPVisualization.png"))
+        plot_dim_reduced_data(embedding[:, 0], embedding[:, 1], fig_id=1,
+                              hue=cell_data[category], cell_data=cell_data, title=graph_title,
+                              save_dir=save_dir, save_file="UMAPVisualization.png")
 
     elif algorithm == "PCA":
         pca = PCA()
         pca_result = pca.fit_transform(cell_data[columns].values)
 
-        fig2 = plt.figure(2)
-        sns.scatterplot(x=pca_result[:, 0], y=pca_result[:, 1], hue=cell_data[category],
-                        palette=color_map, data=cell_data, legend="full", alpha=0.3)
-        plt.legend(bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.)
-        plt.title('PCA projection of data', fontsize=24)
-        fig2.show()
-
-        if save_dir is not None:
-            plt.savefig(os.path.join(save_dir, "PCAVisualization.png"))
+        plot_dim_reduced_data(pca_result[:, 0], pca_result[:, 1], fig_id=2,
+                              hue=cell_data[category], cell_data=cell_data, title=graph_title,
+                              save_dir=save_dir, save_file="PCAVisualization.png")
 
     elif algorithm == "tSNE":
         tsne = TSNE()
         tsne_results = tsne.fit_transform(cell_data[columns].values)
 
-        fig3 = plt.figure(3)
-        sns.scatterplot(
-            x=tsne_results[:, 0], y=tsne_results[:, 1],
-            hue=cell_data[category],
-            palette=color_map,
-            data=cell_data,
-            legend="full",
-            alpha=0.3
-        )
-        plt.legend(bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.)
-        plt.title('tSNE projection of data', fontsize=24)
-        fig3.show()
-
-        if save_dir is not None:
-            plt.savefig(os.path.join(save_dir, "tSNEVisualization.png"))
+        plot_dim_reduced_data(tsne_results[:, 0], tsne_results[:, 1], fig_id=3,
+                              hue=cell_data[category], cell_data=cell_data, title=graph_title,
+                              save_dir=save_dir, save_file="tSNEVisualization.png")
```

Comment:
This can be condensed:
```
func_map = {
    'UMAP' : umap.UMAP,
    'PCA' : PCA,
    'tSNE' : TSNE,
}

reducer = func_map[algorithm]()
results = reducer.fit_transform(
    cell_data[columns].values
    if algorithm != 'UMAP'
    else StandardScaler().fit_transform(cell_data[columns].values)
)

plot_dim_reduced_data(results[:, 0], results[:, 1], fig_id=list(func_map.keys()).index(algorithm),
                      hue=cell_data[category], cell_data=cell_data, title=graph_title,
                      save_dir=save_dir, save_file=f"{algorithm}Visualization.png")
```

Reply:
This is very nice but I think reduces code readability. The proposed changes already dramatically clean things up, I think that's fine

Reply:
Hey, that's pretty good. Gonna need to add some comments in case anyone gets confused by this method lol but it's really good.

Reply:
Ah OK, I'm going to have to do a quick revert with this one.